src/main/java/mops/businesslogic/DirectoryServiceImpl.java

```diff
@@ -0,0 +1,255 @@
+        Optional<Directory> optionalDirectory = directoryRepository.findById(parentDirID);
+        return optionalDirectory.orElseThrow(getException(parentDirID)); //NOPMD// this is not a violation of demeter's law
+    }
+
+
+    /**
+     * Checks if the user has permission to write in that folder.
+     *
+     * @param account   user credentials
+     * @param directory directory object of the permissions requested
+     */
+    private void checkWritePermission(Account account, Directory directory) throws WriteAccessPermission {
+        DirectoryPermissions directoryPermissions = getDirectoryPermissions(directory);
+
+        String userRole = permissionService.fetchRoleForUserInDirectory(account, directory);
+
+        boolean allowedToWrite = directoryPermissions.getPermissions()
+                .stream()
+                .filter(DirectoryPermissionEntry::isCanWrite)
+                .anyMatch(permission -> permission.getRole().equals(userRole));
+
+        if (!allowedToWrite) {
+            throw new WriteAccessPermission(String.format("The user %s doesn't have write access to %s.",
+                    account.getName(),
+                    directory.getName()));
+        }
+    }
+
+    private void checkReadPermission(Account account, Directory directory) throws ReadAccessPermission {
+        DirectoryPermissions directoryPermissions = getDirectoryPermissions(directory);
+
+        String userRole = permissionService.fetchRoleForUserInDirectory(account, directory);
+
+        boolean allowedToWrite = directoryPermissions.getPermissions()
+                .stream()
+                .filter(DirectoryPermissionEntry::isCanRead)
+                .anyMatch(permission -> permission.getRole().equals(userRole));
+
+        if (!allowedToWrite) {
+            throw new ReadAccessPermission(String.format("The user %s doesn't have read access to %s.",
+                    account.getName(),
+                    directory.getName()));
+        }
+
+    }
+
+    private void checkIfAdmin(Account account, Long groupId) throws WriteAccessPermission {
+        String role = permissionService.fetchRoleForUserInGroup(account, groupId);
+        if (!ADMINISTRATOR.equals(role)) {
+            String errorMessage = String.format(
+                    "User is not %s of %d and there for not allowed to create a root folder.",
+                    ADMINISTRATOR,
+                    groupId);
+            throw new WriteAccessPermission(errorMessage);
+        }
+    }
+
+    private DirectoryPermissions getDirectoryPermissions(Directory directory) {
+        Optional<DirectoryPermissions> optDirPerm = directoryPermissionsRepo.findById(directory.getPermissionsId());
+        return optDirPerm.orElseThrow(getException(directory.getId()));
+    }
+
+
+    private Set<DirectoryPermissionEntry> createDefaultPermissions(Set<String> roleNames) {
+        return roleNames.stream() //NOPMD// this is not a violation of demeter's law
```

Comment:
use `@SuppressWarnings`

Reply:
Is that useful if I only want to ignore that line?

Reply:
yes, because the annotation is cleaner than the comment